plugins/modules/vmware_host.py

```diff
@@ -788,6 +793,37 @@ def state_update_host(self):
 
         self.module.exit_json(changed=changed, msg=str(result))
 
+    def state_disconnected_host(self):
+        """Disconnect host to vCenter"""
+        changed = True
+        result = None
+
+        if self.module.check_mode:
+            result = "Host would be disconnected host from vCenter '%s'" % self.vcenter
+        else:
+            if self.host.runtime.connectionState == 'disconnected':
+                changed = False
+                result = "Host already disconnected"
+            else:
+                self.disconnect_host(self.host)
+                result = "Host disconnected from vCenter '%s'" % self.vcenter
```

Comment:
```suggestion
        if self.host.runtime.connectionState == 'disconnected':
            changed = False
            result = "Host already disconnected"
        else:
            changed = True
            if self.module.check_mode:
                result = "Host would be disconnected from vCenter '%s'" % self.vcenter
            else:
                self.disconnect_host(self.host)
                result = "Host disconnected from vCenter '%s'" % self.vcenter
```

It looks like the module would always reports a change when `check_mode` is enabled, even if the host already is disconnected. Wouldn't it be better to not report a change in this case?

Reply:
Thank you @mariolenz for reviewing!  
I fixed that if ESXi has already been disconnected, the changed will not occur with the check_mode option specified.

Reply:
I think your suggestion is simple and good!  
However, I was concerned that omitting the process might confuse the check_mode process, so I modified it to clearly separate the two.